src/main/java/org/elasticsearch/indices/cache/query/IndicesQueryCache.java

```diff
@@ -235,10 +234,11 @@ public QuerySearchResultProvider load(final ShardSearchRequest request, final Se
             }
         } else {
             key.shard.queryCache().onHit();
+            // restore the cached query result into the context
+            final QuerySearchResult result = context.queryResult();
+            result.readFromWithId(context.id(), value.streamInput());
```

Comment:
I think the point of having BytesQuerySearchResult was that we don't have to deserialize QuerySearchResult on the source shard if we get it from cache. I believe this is lost now - was it in the way of something?